### PR TITLE
QgsSymbol::renderFeature(): render larger parts of multipolygon first

### PR DESCRIPTION
**Pull request by rouault**
When drawing an invalid multipolygon, that has a part inside another one,
there is a chance we draw first the smaller part and then the larger part,
making it invisible.
Change the drawing order to start with larger parts.

Fixes #15419

### CODE REVIEW

File: src/core/symbology-ng/qgssymbol.cpp [hunk 1]
```diff
@@ -842,30 +843,49 @@ void QgsSymbol::renderFeature( const QgsFeature& feature, QgsRenderContext& cont
       const QgsGeometryCollection& geomCollection = dynamic_cast<const QgsGeometryCollection&>( *segmentizedGeometry.geometry() );
       const unsigned int num = geomCollection.numGeometries();
 
+      // Sort components by approximate area (probably a bit faster than using
+      // area() )
+      std::map<double, QList<unsigned int> > mapAreaToPartNum;
```
Comment (rouault) [inline src/core/symbology-ng/qgssymbol.cpp]: I needed a reverse iterator and couldn't find one in QMap API, unless I missed it

Comment (m-kuhn) [inline src/core/symbology-ng/qgssymbol.cpp]: I see. I think starting with constEnd() and then going i-- should work.

Comment (rouault) [inline src/core/symbology-ng/qgssymbol.cpp]: This works, but this is super ugly. See

```
     if(  num == 0 )
          break;
      QMap<double, QList<unsigned int> >::const_iterator iter = mapAreaToPartNum.constEnd();
      bool stopAtNextIteration = false;
      --iter;
      while (true) {
        const QList<unsigned int>& listPartIndex = iter.value();
        [...]
        if( stopAtNextIteration )
            break;
        -- iter;
        if( iter == mapAreaToPartNum.constBegin() )
            stopAtNextIteration = true;
      }
```

**Comment by m-kuhn** [comment]
No worries, in this case I'd say code readability matters more than
consistency.

**Pull request merged**